src/security/content-security-policy.conf

```diff
@@ -8,18 +8,69 @@
+#
+#      If you want to use the `<base>` element, then `base-uri 'self'` 
+#      can be used instead.
+#
+#  (2) All resources are restricted to the origin of the current website 
+#      by setting the `default-src` directive to `'self'` - which acts as a 
+#      fallback to all "Fetch directives" (https://developer.mozilla.org/en-US/docs/Glossary/Fetch_directive).
+#
+#      This is conveniant as you do not have to specify all Fetch directives 
+#      that apply to your site, for example: 
+#      `connect-src 'self'; font-src 'self'; script-src 'self'; style-src 'self'`, etc. 
+#
+#      This restriction also means that you must explicitly define from
+#      which site(s) your website is allowed to load resources from.
+#
+#  (3) Scripts are only allowed to be loaded from the current website 
+#      and an example CDN website.
+#  
+#  (4) Form submissions are only allowed from the current website by 
+#      setting: `form-action 'self'`.
+#
+#  (5) Prevents other websites from embedding your website with e.g. the 
+#      `<iframe>` or `<object>` element, by setting `frame-ancestors 'self'`. 
+#
+#	   The `frame-ancestors` directive helps avoid "Clickjacking" attacks
+#      and is similar to the `X-Frame-Options` header.
+#
+#      Browsers that support the CSP header will ignore `X-Frame-Options` 
+#      if `frame-ancestors` is also specified.
+#
+#  (6) Forces the browser to treat all the resources that are served over
+#      HTTP as if they were loaded securely over HTTPS by setting the
+#      `upgrade-insecure-requests` directive.
+#
+#      Please note that `upgrade-insecure-requests` does not ensure 
+#      HTTPS for the top-level navigation. If you want to force the
+#      website itself to be loaded over HTTPS you must include the
+#      `Strict-Transport-Security` header.
+#
+# To make your CSP implementation easier, you can use an online CSP header 
+# generator such as: 
+# https://www.cspisawesome.com/
+# https://report-uri.com/home/generate/.
+#
+# It is encouraged that you validate your CSP header using a CSP validator
+# such as:
+# https://csp-evaluator.withgoogle.com
+#
+# https://csp.withgoogle.com/docs/
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
-# https://www.w3.org/TR/CSP3/
-# https://content-security-policy.com/
 # https://www.html5rocks.com/en/tutorials/security/content-security-policy/
+# https://www.w3.org/TR/CSP/
 
 <IfModule mod_headers.c>
-    Header set Content-Security-Policy "script-src 'self'; object-src 'self'" "expr=%{CONTENT_TYPE} =~ m#text/html#i"
+    #                                   (1)              (2)                 (3)                                        (4)                 (5)                     (6)
+    Header set Content-Security-Policy "base-uri 'none'; default-src 'self'; script-src 'self' https://cdn.example.com; form-action 'self'; frame-ancestors 'self'; upgrade-insecure-requests" "expr=%{CONTENT_TYPE} =~ m#text/html#i"
```

Comment:
* `default-src` first.
* `https://cdn.example.com` is here for the example of course, but I don't like adding weakness for the examples, even if nothing is behind `cdn.example.com`. Also using hosts for `script-src` is not recommended.
* `form-action` duplicates the `default-src`, keeping defaults as default is sufficient. 😉 
* `frame-ancestors` should be set to `none` to match with current h5bp value.

Reply:
> default-src first.

Done!

> Also using hosts for `script-src` is not recommended.

I don't follow, if you want to include a script from another origin then this is the method. <br>However:

> `https://cdn.example.com` is here for the example of course, but I don't like adding weakness for the examples [...]

Removed!

> `frame-ancestors` should be set to `none` to match with current h5bp value. 

Done!

> `form-action` duplicates the `default-src` [...]

`form-action` is a _navigation_ directive, only _fetch_ directives falls back to `default-src`. The default value of `form-action` is `*`.

Reply:
> I don't follow, if you want to include a script from another origin then this is the method.

I was thinking the same thing two days before, but:

> Host whitelists can frequently be bypassed. Consider using 'strict-dynamic' in combination with CSP nonces or hashes.
[csp-evaluator.withgoogle.com](https://csp-evaluator.withgoogle.com/)

That said I didn't verify the source of that displayed info.

> `form-action` is a [_navigation_ directive](https://www.w3.org/TR/CSP/#directives-navigation)

Oh, you're totally right, sorry.

Reply:
> Host whitelists can frequently be bypassed. Consider using `'strict-dynamic'` in combination with CSP nonces or hashes.

This blog post (I didn't read it all yet): https://websec.be/blog/cspstrictdynamic/ seems to go into some detail on insecure whitelists, the main issue seems to be allowing endpoints that serve JSONP. 

`script-src` is easier to implement but it seems you're right. `script-src` is removed from the example policy however.